.github/workflows/ci.yml

```diff
@@ -1,15 +1,34 @@
-name: ci
+name: CI Workflow
 
 on:
   pull_request:
     branches:
       - main
 
-  workflow_dispatch:
-
 jobs:
+  block-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR is from forked repo
+        id: check_forked
+        run: |
+          if [[ ${{ github.event.pull_request.head.repo.full_name }} != ${{ github.event.pull_request.base.repo.full_name }} ]]; then
+            echo "{is_forked}=={true}" >> $GITHUB_OUTPUT
+          else
+            echo "{is_forked}=={false}" >> $GITHUB_OUTPUT
+          fi
+      - name: Block PR from forked repo
```

Comment:
Is there a need for a second step? Can't we consolidate this step into the previous one?